airsonic-main/src/main/java/org/airsonic/player/domain/CacheElement.java

```diff
@@ -41,7 +41,12 @@ public CacheElement(int type, String key, Object value, long created) {
     }
 
     public static long createId(int type, String key) {
-        return ((long) type << 32) | Math.abs(key.hashCode());
+        int hashcode = key.hashCode();
+        // http://findbugs.sourceforge.net/bugDescriptions.html#RV_ABSOLUTE_VALUE_OF_HASHCODE
+        if (hashcode == Integer.MIN_VALUE) {
+          hashcode++;
+        }
+        return ((long) type << 32) | Math.abs(hashcode);
     }
```

Comment:
This could be `return ((long) type << 32) | (key.hashCode() & Integer.MAX_VALUE);`